### PR TITLE
Add support for SubRip

### PR DESCRIPTION
**Pull request by joksas**
The syntax is described [here](https://docs.fileformat.com/video/srt/).

MIME types `text/x-subrip` and `text/x-srt` were suggested [here](https://github.com/gabriel-vasile/mimetype/pull/222#pullrequestreview-836628591), while the type `application/srt` is motivated by [this](https://rss-extension.buzzsprout.com/1.0/podcast/).

### CODE REVIEW

**Comment by joksas** [comment]
Huh, that's strange -- it works on Linux, but not on Windows. I'll take a look at that.

**Comment by joksas** [comment]
Alright, it works now.

**Comment by gabriel-vasile** [comment]
Hi,
I added a benchmark for .srt detection with regex vs. lines parsing.
On my machine parsing lines is almost 10 times faster and I think this is enough to justify not using regex for this.
```bash
cd internal/magic
go test -bench=.

goos: linux
goarch: amd64
pkg: github.com/gabriel-vasile/mimetype/internal/magic
cpu: Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz
BenchmarkRegexSrt-8   	  160696	      7496 ns/op
BenchmarkParseSrt-8   	 1370484	       824.4 ns/op
PASS
ok  	github.com/gabriel-vasile/mimetype/internal/magic	4.368s
```
I didn't test it on windows, but it should work too. I'm not sure there are no other bugs. Tell me if you see anything wrong with it.

**Review by gabriel-vasile — COMMENTED** [review]
Hi,
I added a benchmark for .srt detection with regex vs. lines parsing.
On my machine parsing lines is almost 10 times faster and I think this is enough to justify not using regex for this.
```bash
cd internal/magic
go test -bench=.

goos: linux
goarch: amd64
pkg: github.com/gabriel-vasile/mimetype/internal/magic
cpu: Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz
BenchmarkRegexSrt-8   	  160696	      7496 ns/op
BenchmarkParseSrt-8   	 1370484	       824.4 ns/op
PASS
ok  	github.com/gabriel-vasile/mimetype/internal/magic	4.368s
```
I didn't test it on windows, but it should work too. I'm not sure there are no other bugs. Tell me if you see anything wrong with it.

**Comment by joksas** [comment]
Hi, thank you a lot for this!

Yes, I agree that it makes sense to use parsing. I only made the checks on timestamps a bit more restrictive.

Here are the benchmarks before I made the changes (i.e. your version):
```
cpu: AMD Ryzen 5 2600 Six-Core Processor
BenchmarkRegexSrt-12           46358         25832 ns/op
BenchmarkParseSrt-12          397768          2912 ns/op
```

and after my changes:
```
cpu: AMD Ryzen 5 2600 Six-Core Processor
BenchmarkRegexSrt-12           45734         25546 ns/op
BenchmarkParseSrt-12          394863          3043 ns/op
```
So the increase in time is only marginal (if statistically significant at all).

Anyway, sorry for merging too early, I just didn't expect that the function would work with Go 1.17.6 (my local version) and fail with Go 1.16.12 (GitHub Actions version)... I did confirm that this is the case by installing the latter version locally. I'll try to find out what's happening.

**Comment by joksas** [comment]
Alright, it seems that comma as a decimal separator for fractional seconds was only introduced in [this commit](https://github.com/golang/go/commit/f02a26be). I'll think about the best way around this.

**Comment by joksas** [comment]
It now works with Go versions less than 1.17 as well.

As a side note, would it make sense to standardise the package's minimum Go version defined in [go.mod](https://github.com/gabriel-vasile/mimetype/blob/master/go.mod) and the version used by [GitHub actions](https://github.com/gabriel-vasile/mimetype/blob/master/.github/workflows/go.yml)? The former is 1.12 and the latter is 1.16. I don't think it matters that much what specific version is used, only that it is consistent throughout the project. Tests passing in 1.16 does not necessarily imply that they will pass in 1.12.

File: internal/magic/text.go [hunk 1]
```diff
@@ -297,3 +299,47 @@ func HAR(raw []byte, limit uint32) bool {
...
+	}
+	// First line must be 1.
+	if s.Text() != "1" {
+		return false
+	}
+
+	if !s.Scan() {
+		return false
+	}
+	secondLine := s.Text()
```
Comment (gabriel-vasile) [inline internal/magic/text.go:317]:
```suggestion
	secondLine := s.Text()
	// Timestamp format (e.g: 00:02:16,612 --> 00:02:19,376) limits secondLine
	// length to exactly 29 characters.
	if len(secondLine) != 29 {
		return false
	}
```

This change prevents `strings.Contains` and `strings.ReplaceAll` from unnecessary running and allocating memory. Example of such input:
```
1
insanely long line here, unrelated to the srt format...
```

**Review by gabriel-vasile — COMMENTED** [review]
Hi,
Code looks good, nice catch finding the comma issue.
I left one last suggestion.

**Comment by joksas** [comment]
The suggestion makes sense, I've committed it -- thank you.

**Review by gabriel-vasile — COMMENTED** [review]
One last last problem. I thought about it, forgot to write it down and only noticed it now, at the last review.

`application/srt` is not a valid MIME type because it's in the [standards tree](https://en.wikipedia.org/wiki/Media_type#Standards_tree) but it is not registered at [IANA](https://www.iana.org/assignments/media-types/media-types.xhtml).
buzzsprout.com cannot just decide to use `application/srt` by themselves. They need to follow the registration process with IANA or use one of the [vendor tree](https://en.wikipedia.org/wiki/Media_type#Vendor_tree) or the [unregistered tree](https://en.wikipedia.org/wiki/Media_type#Unregistered_tree).

The other MIME types (`text/x-subrip` and `text/x-srt`) are ok. They are experimental (because of the `x-` part) and don't need registration with IANA to be used, but `application/srt` shouldn't be here.

**Comment by joksas** [comment]
Got it. I've removed `application/srt`.

**Review by gabriel-vasile — APPROVED** [review]

**Pull request merged**